src/components/info/location-info.vue

```diff
@@ -0,0 +1,193 @@
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+import { useTranslation } from 'i18next-vue'
+import { storeToRefs } from 'pinia'
+import { useInfoStore } from '@/stores/info.store'
+import { Coordinate } from 'ol/coordinate'
+import useMap from '@/composables/map/map.composable'
+import {
+  getQRUrl,
+  getElevation,
+  getNearestAddress,
+  queryInfo,
+} from '@/services/info/location-info'
+
+import StreetView from '@/components/info/street-view.vue'
+
+const { t } = useTranslation()
+const { locationInfo } = storeToRefs(useInfoStore())
+
+const map = useMap().getOlMap()
+
+const shortUrl = ref()
+const qrUrl = ref()
+const elevation = ref()
+const address = ref()
+const clickCoordinateLuref = ref()
+const formatted_coordinates = ref({})
+
+// initial load of position infos if locationInfo is not initially undefined
+// async function, no need to await the completion, DOM will be updated via refs
+updateInfo(locationInfo.value)
+
+watch(locationInfo, updateInfo)
+
+async function updateInfo(location: Coordinate | undefined) {
+  if (location) {
+    const info = await queryInfo(location, map.getView().getProjection())
+    shortUrl.value = info.shortUrl
+    qrUrl.value = getQRUrl(info.shortUrl)
+    clickCoordinateLuref.value = info.clickCoordinateLuref
+    formatted_coordinates.value = info.formatted_coordinates
+    elevation.value = await getElevation(info.clickCoordinateLuref)
+    address.value = await getNearestAddress(info.clickCoordinateLuref)
+  }
+}
+
+const isRapportForageVirtuelAvailable = computed(() => {
+  const userRole = 'ACT'
+  return userRole === 'ACT'
+})
+const downloadingRepport = ref(false)
+const isInBoxOfLidar = computed(() => true)
+const isCyclomediaAvailable = computed(() => true)
+const isImagesObliquesAvailable = computed(() => true)
+
+const getLidarUrl = () => 'bla'
+const cyclomediaUrl = computed(() =>
+  clickCoordinateLuref.value
+    ? `http://streetsmart.cyclomedia.com/streetsmart?q=${clickCoordinateLuref.value[0]};${clickCoordinateLuref.value[1]}`
+    : ''
+)
+const imagesObliquesUrl = computed(() =>
+  clickCoordinateLuref.value
+    ? `https://oblique.geoportail.lu/publication/viewer?x=${clickCoordinateLuref.value[0]}&y=${clickCoordinateLuref.value[1]}&crs=2169`
+    : ''
+)
+
+const addRoutePoint = () => '// TODO: add point to route'
+
+const open = ref(true)
+const { isStreetviewActive } = storeToRefs(useInfoStore())
+const toggleStreetview = () => {
```

Comment:
```suggestion
function toggleStreetview () {
```

Any reason why this is declared as a const?

Reply:
is there any difference ?

Reply:
Code consistency?